src/App.tsx

```diff
@@ -1,8 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+import * as api from './api';
+import PostsList from './PostList';
+
+const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const loadData = async () => {
+    setIsLoading(true);
+
+    const postsFromServer = await api.getPosts();
+
+    setPosts(postsFromServer);
+    setIsLoading(false);
+    setIsLoaded(true);
+  };
+
+  return (
+    <>
+      <h1>Dynamic list of posts</h1>
+      {isLoaded ? (
+        <PostsList posts={posts} />
+      ) : (
+        <button
+          type="button"
+          onClick={loadData}
+        >
+          {isLoading ? (
```

Comment:
it is difficult to quickly understand logic if there are nested ternary operators in ternary operators.

the best option to simplify the code is to create a separate constant for the button title value

```javascript
 const buttonTitle = isLoading ? 'Loading...' : 'Load'
 
 ________________________________________________
 
 <button onClick={loadData}  disabled={isLoading} >{buttonTitle}</button>
 
 
```